lib/ansible/modules/replace.py

```diff
@@ -284,21 +284,27 @@ def main():
         section = contents
 
     mre = re.compile(params['regexp'], re.MULTILINE)
-    result = re.subn(mre, params['replace'], section, 0)
-
-    if result[1] > 0 and section != result[0]:
-        if pattern:
-            result = (contents[:indices[0]] + result[0] + contents[indices[1]:], result[1])
-        msg = '%s replacements made' % result[1]
-        changed = True
-        if module._diff:
-            res_args['diff'] = {
-                'before_header': path,
-                'before': contents,
-                'after_header': path,
-                'after': result[0],
-            }
-    else:
+    try:
```

Comment:
This `try/except` wraps too much code, and I believe it wraps code not affected by the exception.

We should just be wrapping the `re.compile` and if that fails, we should immediately call `module.fail_json` regardless of the error.